tests/test_builder.py

```diff
@@ -42,6 +43,56 @@ def test_build_runtime_use_image_when_no_build():
     assert fn.spec.image == image
 
 
+def test_build_runtime_insecure_registries(monkeypatch):
+    get_k8s_helper_mock = unittest.mock.Mock()
+    monkeypatch.setattr(
+        mlrun.builder, "get_k8s_helper", lambda *args, **kwargs: get_k8s_helper_mock
+    )
+    mlrun.builder.get_k8s_helper().create_pod = unittest.mock.Mock(
+        side_effect=lambda pod: (pod, "some-namespace")
+    )
+    mlrun.mlconf.httpdb.builder.docker_registry = "registry.hub.docker.com/username"
+    function = mlrun.new_function(
+        "some-function",
+        "some-project",
+        "some-tag",
+        image="mlrun/mlrun",
+        kind="job",
+        requirements=["some-package"],
+    )
+
+    insecure_flags = {"--insecure", "--insecure-pull"}
+    for pull_mode, push_mode, secret, flags_expected in (
+        ("auto", "auto", "", True),
+        ("auto", "auto", "some-secret-name", False),
+        ("enabled", "enabled", "some-secret-name", True),
+        ("enabled", "enabled", "", True),
+        ("disabled", "disabled", "some-secret-name", False),
+        ("disabled", "disabled", "", False),
+    ):
```

Comment:
If we want to add another attribute to the test cases in the future, since you are currently looping over tuples, you will need to add the attribute in each tuple so the loop would not fail.
We often use test cases as dictionaries, which allows for greater explainability and the ability to set only the attributes that are required for a specific test case.

What do you think?

Reply:
Yalla